### PR TITLE
refactor: move chain ID prepending from provers to Inbox contract

### PR DESCRIPTION
**Pull request by re1ro**
## Summary
- Moved chain ID prepending logic from individual prover implementations to the Inbox contract
- Centralized the chain ID handling to reduce code duplication across provers
- Updated all tests to account for the 8-byte chain ID prefix in proof messages

## Changes
- Added `CHAIN_ID` immutable variable in `Inbox.sol` to store and prepend chain ID to encoded proofs
- Removed `CHAIN_ID` storage and prepending logic from `MessageBridgeProver` and all inheriting provers (HyperProver, LayerZeroProver, MetaProver)
- Removed redundant proof length validation checks from individual provers since the format is now standardized
- Updated test contracts and test suites to handle the new message format with chain ID prefix

## Test Plan
- [x] All existing tests pass with updated proof encoding format
- [x] Edge case tests properly handle chain ID prefix
- [x] Test contracts correctly parse proofs with 8-byte chain ID offset

### CODE REVIEW

File: contracts/prover/BaseProver.sol [hunk 1]
```diff
@@ -51,7 +51,7 @@ abstract contract BaseProver is IProver, ERC165 {
 
     /**
      * @notice Process intent proofs from a cross-chain message
-     * @param data Encoded (intentHash, claimant) pairs
+     * @param data Encoded (intentHash, claimant) pairs (without chain ID prefix)
```
Comment (fish-sammy) [inline contracts/prover/BaseProver.sol:54]: 👍

File: contracts/prover/MetaProver.sol [hunk 2]
```diff
@@ -230,11 +230,6 @@ contract MetaProver is IMetalayerRecipient, MessageBridgeProver, Semver {
         view
         returns (uint32 domain, bytes32 recipient, bytes memory message)
     {
```
[the remark this answers is not in the document]
Comment (re1ro) [reply contracts/prover/MetaProver.sol:232]: can't be pure because we use CHAIN_ID, immutable

File: contracts/prover/LayerZeroProver.sol [hunk 3]
```diff
@@ -245,11 +245,6 @@ contract LayerZeroProver is ILayerZeroReceiver, MessageBridgeProver, Semver {
         view
         returns (ILayerZeroEndpointV2.MessagingParams memory params)
     {
```
[the remark this answers is not in the document]
Comment (fish-sammy) [reply contracts/prover/LayerZeroProver.sol:247]: WHAT?! Where do we use it

[the remark this answers is not in the document]
Comment (nknavkal) [reply contracts/prover/LayerZeroProver.sol:247]: we used to use it but you took it out right? should be good to purify
